hazelcast/include/hazelcast/client/protocol/ClientMessage.h

```diff
@@ -1122,6 +1122,10 @@ class HAZELCAST_API ClientMessage
             if (is_final) {
                 h->flags |= IS_FINAL_FLAG;
             }
+
+            if (std::is_same<T, serialization::pimpl::data>::value) {
```

Comment:
If this is a better approach than what Java does let's also send a PR to Java. Java uses the generated codec to set this field explicitly the and the codec generator decides it. Your approach also seems to be appropriate and doable by Java, can you discuss with the Java team about it? Let's unify both approaches with whichever is better.

Reply:
I already talked with @mdumandag but we didn't change Java side because it is already there and working.